### DISCUSSION TITLE
Where does one find the command class numbers?

### DISCUSSION THREAD

**Discussion by TheKorn2** [zwave-js/zwave-js-ui#1196] [Q&A] [answered]
In the API section of the docs, for sendCommand, the ctx component says it needs three things...  the node ID, the endpoint, and the command class number.

Where does one find a mapping of command classes <--> command class numbers?

Let's say I'm trying to do a basic set. So node ID would be 2, endpoint would be 0, and command class number would be....  ??????

And what would the command argument be?  BasicCCSet? BasicCC? Set?  Something else my lizard brain can't quite grasp yet?

Thanks in advance!

**Comment by robertsLando** [comment, marked as answer]
I have asked to @AlCalzone to add them in his [docs](https://zwave-js.github.io/node-zwave-js/#/api/CCs/index). In the meanwhile you could find them [here](https://www.silabs.com/documents/login/miscellaneous/SDS13548-List-of-defined-Z-Wave-Command-Classes.xlsx)

**Reply by AlCalzone** [reply]
https://github.com/zwave-js/node-zwave-js/issues/2611

**Reply by TheKorn2** [reply]
OK!  One step forward, thanks!

But I'm trying to match up the example, and the effort is raising more questions than it's answering.  The example of a sent command in the docs is startLevelChange, which isn't listed in either the sigma documentation you linked nor the zwavejs docs.

So where did startLevelChange come from? What command class does it belong to? (0x38 isn't in the sigma docs.) And is there a listing of *those* names anywhere???

Sorry for the thousand questions, probably just a fresh eyes type problem.

**Reply by AlCalzone** [reply]
It definitely is in the official documentation. It isn't `0x38` (hex), but `38` decimal, which is `0x26` in hex:
```
COMMAND_CLASS_SWITCH_MULTILEVEL | Application | Active | 0x26
```
```
Multilevel Switch Command Class:
	Multilevel Switch Set
	Multilevel Switch Get
	Multilevel Switch Report
	Multilevel Switch Start Level Change
	Multilevel Switch Stop Level Change
	Multilevel Switch Supported Get
	Multilevel Switch Supported Report
```

@robertsLando I think it would be good if the mqtt API not only accepted the CC IDs (numbers) but also their names as defined in `node-zwave-js`. So for a `startLevelChange` the command could look like this:
```
{ "args": [
  {
    "nodeId": 23,
    "commandClass": "Multilevel Switch",
    "endpoint": 0,
  },
  "startLevelChange",
]}
```

**Reply by robertsLando** [reply]
I agree!

**Reply by TheKorn2** [reply]
@AlCalzone Ahhhh, thanks! I was looking at the TOC of the command classes api. It didn't click that the example was one level down. Makes sense now that I think about it; you wouldn't ever send a "multilevel switch" command per se, as that doesn't say to _do_ anything.

**Reply by robertsLando** [reply]
Cpuld you open a feature requeest for this? @TheKorn2

**Reply by TheKorn2** [reply]
@robertslando  sure, np

done
